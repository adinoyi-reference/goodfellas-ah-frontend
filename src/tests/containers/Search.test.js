import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Search, { SearchArticlesContainer } from '../../containers/SearchArticles';
import Root from '../../root';

let wrapper;
// let wrapped;
let searchMount;
// let mockGetArticles;

const articles = {
  articles: {
    searchResults: {
      success: true,
      articles: [
        {
          favorited: null,
          favoritesCount: null,
          id: 2,
          slug: 'jake-peralta-is-an-idiot',
          title: 'Lorem ipsum, please teach me to right',
          description: 'Stressss',
          body: 'Do you know what the perimeter of the earth is? like the whole earth o, including the ocean? yh i don\'t know it either. ',
          image: 'https://res.cloudinary.com/drmmqcxkc/image/upload/v1541610101/v6secq0grmfsayjb22wt.jpg',
          tagList: null,
          read_time: '1 minute',
          averageRating: null,
          authorId: 1,
          createdAt: '2018-11-07T16:48:19.350Z',
          updatedAt: '2018-11-07T17:01:42.065Z',
          user: {
            firstname: 'Jeff',
            lastname: 'Britta',
            profile: {
              image: 'https://res.cloudinary.com/drmmqcxkc/image/upload/v1541609731/sa6kvtumxdmsdwyz9xcj.jpg',
            },
          },
          reactions: [
            {
              reaction: 1,
            },
            {
              reaction: 1,
            },
          ],
          comments: [],
        },
        {
          favorited: null,
          favoritesCount: null,
          id: 3,
          slug: 'horsin-around',
          title: 'Horsin\' around!',
          description: 'Stressss',
          body: 'Today, we have with us the star of the 90s show Horsin\' around. Bojack Horseman!. Hello Bojack. How is it going today? ',
          image: null,
          tagList: null,
          read_time: '1 minute',
          averageRating: null,
          authorId: 1,
          createdAt: '2018-11-07T16:48:43.392Z',
          updatedAt: '2018-11-07T16:48:43.392Z',
          user: {
            firstname: 'Jeff',
            lastname: 'Britta',
            profile: {
              image: 'https://res.cloudinary.com/drmmqcxkc/image/upload/v1541609731/sa6kvtumxdmsdwyz9xcj.jpg',
            },
          },
          reactions: [],
          comments: [],
        },
      ],
    },
    searchError: {
      data: {
        success: false,
        message: 'We couldn\'t find any articles.',
      },
      status: 404,
      statusText: 'Not Found',
      headers: {
        'content-type': 'application/json; charset=utf-8',
      },
      config: {
        transformRequest: {},
        transformResponse: {},
        timeout: 0,
        xsrfCookieName: 'XSRF-TOKEN',
        xsrfHeaderName: 'X-XSRF-TOKEN',
        maxContentLength: -1,
        headers: {
          Accept: 'application/json, text/plain, */*',
        },
        method: 'get',
        url: 'http://localhost:5000/api/articles/search?article=home&author=false&tag=false',
      },
      request: {},
    },
  },
};
beforeEach(() => {
  wrapper = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
        <Search />
      </MemoryRouter>
    </Root>,
  );

  searchMount = mount(
    <SearchArticlesContainer
      searchResults={articles.articles.articles}
      searchError={articles.error}
    />,
  );
});


afterEach(() => wrapper.unmount());

describe('Search UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});


describe('Search Functionality', () => {
  it('should fail to find articles', () => {
    searchMount.setState({
      articleResults: [articles.articles.searchResults],
      errorResults: [],
    });
    const inst = searchMount.instance();
    expect(inst).toBeInstanceOf(SearchArticlesContainer);
    expect(inst).not.toBeNull();
  });

  it('should find articles', () => {
    searchMount.setState({
      articleResults: [articles.articles.searchResults],
      errorResults: [articles.articles.searchError],
    });
    const inst = searchMount.instance();
    expect(inst).toBeInstanceOf(SearchArticlesContainer);
    expect(inst).not.toBeNull();
  });

  it('should trigger ComponentWillReceiveProps articles', () => {
    searchMount.setProps({
      searchResults: [articles.articles.searchResults],
      searchError: [articles.articles.searchError],
    });
  });

  it('should trigger ComponentWillReceiveProps articles', () => {
    searchMount.setProps({
      searchResults: [articles.articles.searchResults],
      searchError: [undefined],
    });
  });
});
