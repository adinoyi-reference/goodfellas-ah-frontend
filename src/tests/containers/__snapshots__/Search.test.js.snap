// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(SearchArticlesContainer)>
          <SearchArticlesContainer
            dispatch={[Function]}
            searchError={Array []}
            searchResults={Array []}
          >
            <section
              className="search"
            >
              <div>
                <LargeLoader>
                  <div
                    className="large-spinner-wrapper"
                  >
                    <img
                      alt=""
                      className="large-spinner"
                      src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541701878/large-spinner.gif"
                    />
                  </div>
                </LargeLoader>
              </div>
            </section>
          </SearchArticlesContainer>
        </Connect(SearchArticlesContainer)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
